### PR TITLE
Feature validatemail

### LINKED ISSUE #11: 用户注册邮箱验证
- [x] 本地测试
- [x] 集成Mailgun

### PR DESCRIPTION
**Pull request by summerhack**
close #11  waiting for mailgun test

### CODE REVIEW

File: server/maillayout/activemail.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const activemail = link => {
+  return (
```
Comment (kingdido999) [inline server/maillayout/activemail.js]: Leverage ES6 [Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead.

File: server/entities/user/resolvers.js [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
 import userApi from './api'
+import mail from '../../mail'
+import bcrypt from 'bcrypt'
+import config from '../../../config'
+import { GraphQLError } from 'graphql/error'
+import activemail from '../../maillayout/activemail'
 
+const DOMAIN = config.isLocal ? 'http://localhost:3000' : config.domain
+const SALT_WORK_FACTOR = 1
+class newError extends Error {
```
[the remark this answers is not in the document]
Comment (summerhack) [reply server/entities/user/resolvers.js]: use domain&&salt_work_factor for activationcode and activationlink.
deleted newerror.

File: src/pages/ReSendMail/ReSendMail.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Section from '../../components/Section'
+import SendMailForm from '../../containers/SendMailForm'
+import Container from '../../components/Container'
+
+const Login = () => {
```
[the remark this answers is not in the document]
Comment (summerhack) [reply src/pages/ReSendMail/ReSendMail.js]: Done

File: server/entities/user/model.js [hunk 4]
```diff
@@ -29,13 +29,18 @@ const userSchema = mongoose.Schema({
   },
   role: { type: String, default: 'user' }, // ['admin', 'moderator', 'user']
   first_name: String,
-  last_name: String
+  last_name: String,
+  //for mail active
+  is_active: { type: Number, default: 0 },
```
[the remark this answers is not in the document]
Comment (summerhack) [reply server/entities/user/model.js]: Done

File: server/entities/user/model.js [hunk 5]
```diff
@@ -29,13 +29,18 @@ const userSchema = mongoose.Schema({
...
+  active_code: String,
```
[the remark this answers is not in the document]
Comment (summerhack) [reply server/entities/user/model.js]: Done

File: server/entities/user/model.js [hunk 6]
```diff
@@ -29,13 +29,18 @@ const userSchema = mongoose.Schema({
...
+  active_deadline: Date
```
[the remark this answers is not in the document]
Comment (summerhack) [reply server/entities/user/model.js]: Done

File: server/maillayout/activemail.js (hunk 1 above)
Comment (summerhack) [reply server/maillayout/activemail.js]: Done

**Pull request merged**